From 57018000210c578b6de2ecbfca12f59fb6b803c4 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Mon, 05 Oct 2020 06:15:22 +0300
Subject: [PATCH] [SW-21330] goya2: pcie: reset handling

as part of pcie hot-reset, LTSSM enable needs to be masked

Change-Id: Id9e271346ab07d590e8a897f769b0dbff3227b30
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/goya2/common.c b/goya2/common.c
index 9f698dc..0ed61e7 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -298,3 +298,9 @@
 {
 	return mmCPU_TIMESTAMP_CNTCR;
 }
+
+void pci_dev_specific_hot_rst(void)
+{
+	regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
+	regmap_writel(3, mmPCIE_AUX_LTSSM_EN);
+}
diff --git a/goya2_arc_gnu/common.c b/goya2_arc_gnu/common.c
index 1a3273e..aabd318 100644
--- a/goya2_arc_gnu/common.c
+++ b/goya2_arc_gnu/common.c
@@ -186,3 +186,9 @@
 {
 	return mmCPU_TIMESTAMP_CNTCR;
 }
+
+void pci_dev_specific_hot_rst(void)
+{
+	regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
+	regmap_writel(3, mmPCIE_AUX_LTSSM_EN);
+}
